operator/internal/manifests/build.go

```diff
@@ -151,14 +151,16 @@ func ApplyTLSSettings(opts *Options, profile *openshiftconfigv1.TLSSecurityProfi
 		tlsSecurityProfile = profile
 	}
 
-	profileSpec, ok := openshiftconfigv1.TLSProfiles[tlsSecurityProfile.Type]
-
-	if !ok {
-		return kverrors.New("unable to determine tls profile settings")
-	}
-
+	var profileSpec *openshiftconfigv1.TLSProfileSpec
 	if tlsSecurityProfile.Type == openshiftconfigv1.TLSProfileCustomType && tlsSecurityProfile.Custom != nil {
 		profileSpec = &tlsSecurityProfile.Custom.TLSProfileSpec
+	} else {
+		var ok bool
+		profileSpec, ok = openshiftconfigv1.TLSProfiles[tlsSecurityProfile.Type]
+
+		if !ok {
+			return kverrors.New("unable to determine tls profile settings")
+		}
```

Comment:
We could simplify this function like this, WDYT?
```go
// ApplyTLSSettings manipulates the options to conform to the
// TLS profile specifications
func ApplyTLSSettings(opts *Options, profile *openshiftconfigv1.TLSSecurityProfile) error {
	if profile == nil {
		profile = &openshiftconfigv1.TLSSecurityProfile{
			Type: openshiftconfigv1.TLSProfileIntermediateType,
		}
	}

	var (
		minTLSVersion openshiftconfigv1.TLSProtocolVersion
		ciphers       []string
	)

	switch profile.Type {
	case openshiftconfigv1.TLSProfileCustomType:
		if profile.Custom == nil {
			return kverrors.New("missing TLS custom profile spec")
		}
		minTLSVersion = profile.Custom.MinTLSVersion
		ciphers = profile.Custom.Ciphers
	case openshiftconfigv1.TLSProfileOldType, openshiftconfigv1.TLSProfileIntermediateType, openshiftconfigv1.TLSProfileModernType:
		spec := openshiftconfigv1.TLSProfiles[profile.Type]
		minTLSVersion = spec.MinTLSVersion
		ciphers = spec.Ciphers
	default:
		return kverrors.New("unable to determine tls profile settings")
	}

	// need to remap all ciphers to their respective IANA names used by Go
	opts.TLSProfile = TLSProfileSpec{
		MinTLSVersion: string(minTLSVersion),
		Ciphers:       crypto.OpenSSLToIANACipherSuites(ciphers),
	}

	return nil
}
```

Reply:
The only thing I might say here is use another variable other than the function's argument to not change the original content for the profile. Otherwise though, I agree with this.

Reply:
I don't see how replacing an if-else with a whole switch would be simplification. Plus the use of the func param (which is a pointer btw, so changes persist outside the func) doesn't seem wise to me.

Reply:
The `switch` statement is faster to read in my opinion and covers the case where we have identified the `custom` spec, but the spec was not defined. Though I agree that `profile` should not be modified in this context.